src/pytesseract.py

```diff
@@ -192,6 +191,20 @@ def file_to_dict(tsv, cell_delimiter, str_col_idx):
     return result
 
 
+def osd_to_dict(osd):
+    osd_keys = {
+        'Page number': ('page_num', int),
+        'Orientation in degrees': ('orientation', int),
+        'Rotate': ('rotate', int),
+        'Orientation confidence': ('orientation_conf', float),
+        'Script': ('script', str),
+        'Script confidence': ('script_conf', float)
+    }
+    lines = [line.split(': ') for line in osd.split('\n')]
```

Comment:
Instead of using list comprehension, we can use generator expression by switching the [] brackets to ().
That way we will pass the lines on demand (one at a time) instead of creating a whole new copy in the form of a list.
In that way, we will save some memory resources.

Reply:
Of course! Absolutely agree, it's more efficient.